# 1. Add the TBO category

In [1]:
from interface import *

In [2]:
dict_tbo_categories = import_tbo_categories()

In [3]:
pf = open('pickle_data/all_items_info.p', 'rb')
items_info = pickle.load(pf)
pf.close()

In [4]:
def get_categories(item_info):
    metricCategoryID = item_info['metricCategoryID']
    # Delete the first and last strings
    categories = metricCategoryID.split('_')[1:-1]
    return categories

In [6]:
set_tbo = set()
count = 0
indexes_to_delete = []
for i in range(len(items_info)):
    categories = get_categories(items_info[i])
    for category in categories:
        if category in dict_tbo_categories['top']:
            set_tbo.add('top')
        if category in dict_tbo_categories['bottom']:
            set_tbo.add('bottom')
        if category in dict_tbo_categories['onepiece']:
            set_tbo.add('onepiece')
    if len(set_tbo) == 1: # Dealing with items with only one TBO category.
        items_info[i]['tbo'] = list(set_tbo)[0]
        count = count + 1
    else:
        indexes_to_delete.append(i)
    set_tbo.clear()
print "count items with 1 TBO:", count

indexes_to_delete.reverse() # Make the indexes in decreasing order. 
for i in indexes_to_delete:
    del items_info[i]

print "len(items_info):", len(items_info)

count items with 1 TBO: 5108
len(items_info): 5108


### Test

In [7]:
import random

In [45]:
random.choice(items_info)

{'color': 'Dark denim blue rugged rinse',
 'image': {'item': 'http://lp.hm.com/hmprod?set=key[source],value[/model/2016/D00 0399136 004 97 4998.jpg]&set=key[rotate],value[]&set=key[width],value[]&set=key[height],value[]&set=key[x],value[]&set=key[y],value[]&set=key[type],value[STILL_LIFE_FRONT]&set=key[hmver],value[1]&call=url[file:/product/large]',
  'model': 'http://lp.hm.com/hmprod?set=key[source],value[/environment/2016/8AQ_0096_013R.jpg]&set=key[rotate],value[0]&set=key[width],value[4306]&set=key[height],value[5034]&set=key[x],value[485]&set=key[y],value[87]&set=key[type],value[FASHION_FRONT]&set=key[hmver],value[0]&call=url[file:/product/large]'},
 'metricCategoryID': 'LADIES_JEANS_SHAPING_W198',
 'name': 'Shaping Skinny Regular Jeans',
 'serial': '47463-D',
 'tbo': 'bottom',
 'url': 'http://www.hm.com/us/product/47463?article=47463-D',
 'who': 'Women'}

### Serialization

In [9]:
import pickle
f = open('pickle_data/items_info_with_tbo.p', 'wb')
pickle.dump(items_info, f)
f.close()

# 2. Count how many items to extract

In [10]:
from interface import *

In [11]:
dict_tbo_categories = import_tbo_categories()

In [12]:
def get_categories(item_info):
    metricCategoryID = item_info['metricCategoryID']
    # Delete the first and last strings
    categories = metricCategoryID.split('_')[1:-1]
    return categories

In [13]:
def count_has_in(categories, items_info, who):
    # who: 'Women', or 'Men'
    count = 0
    for item_info in items_info:
        if who != item_info['who']:
            continue
        metricCategories = get_categories(item_info)
        for category in categories:
            if category in metricCategories:
                count = count + 1
                break
    return count

### Women

In [15]:
tbo_categories = dict_tbo_categories['top'] + dict_tbo_categories['bottom'] + dict_tbo_categories['onepiece']
women = count_has_in(tbo_categories, items_info, 'Women') 
print women

3712


### Men

In [16]:
men = count_has_in(tbo_categories, items_info, 'Men')
print men

1396


### Women + Men

In [17]:
print men + women

5108


### Women Top

In [18]:
women_top = count_has_in(dict_tbo_categories['top'], items_info, 'Women')
print women_top

1886


### Men Top

In [19]:
men_top = count_has_in(dict_tbo_categories['top'], items_info, 'Men')
print men_top

925


### Women Top + Men Top

In [20]:
print women_top + men_top

2811


### Women Bottom

In [24]:
women_bottom = count_has_in(dict_tbo_categories['bottom'], items_info, 'Women')
print women_bottom

1003


### Men Bottom

In [25]:
men_bottom = count_has_in(dict_tbo_categories['bottom'], items_info, 'Men')
print men_bottom

471


### Women Bottom + Men Bottom

In [26]:
print women_bottom + men_bottom

1474


### Women Onepiece

In [27]:
women_onepiece = count_has_in(dict_tbo_categories['onepiece'], items_info, 'Women')
print women_onepiece

823


### Men Onepiece

In [28]:
men_onepiece = count_has_in(dict_tbo_categories['onepiece'], items_info, 'Men')
print men_onepiece

0


### Women Onepiece + Men Onepiece

In [29]:
print women_onepiece + men_onepiece

823


### Check

In [30]:
women == (women_top + women_bottom + women_onepiece)

True

In [31]:
men == (men_top + men_bottom + men_onepiece)

True

### Result
* The number of items in the TBO categories: 5108 (Women: 3712, Men: 1396)
  * The number of tops: 2811 (Women: 1886, Men: 925)
  * The number of bottoms: 1474 (Women: 1003, Men: 471)
  * The number of onepieces: 823 (Women: 823, Men: 0)